### PR TITLE
WIP: build macOS

### PR DESCRIPTION
**Pull request by hannesa2**

### CODE REVIEW

**Comment by hannesa2** [comment]
@ravinderjangra 
Sorry, I've absolute no clue about typescript, and don't know how to make it run

```
src/ndk.ts(10,18): error TS1005: ',' expected.
src/ndk.ts(10,28): error TS1005: ';' expected.
src/ndk.ts(12,5): error TS1128: Declaration or statement expected.
src/ndk.ts(15,3): error TS1128: Declaration or statement expected.
```

File: dist/index.js [hunk 1]
```diff
@@ -975,6 +975,21 @@ const exec = __importStar(__webpack_require__(986));
 class Ndk {
     constructor(path) {
         this.path = path;
+
+//        var OSName="Unknown OS";
```
Comment (hannesa2) [inline dist/index.js:979]: This PR is just a try out.

Here I run into 
image

Comment (ravinderjangra) [reply dist/index.js:979]: the index.js file is auto-generated from the code `npm run all` and doesn't need to be updated manually.

Comment (hannesa2) [reply dist/index.js:979]: Good to know !
But why add this file to git, when it's auto-generated. Why not remove from git ?

Comment (ravinderjangra) [reply dist/index.js:979]: It seems the GHA directly uses dist folder: https://github.com/actions/typescript-action#publish-to-a-distribution-branch

**Pull request closed without merging**